sdk/provisioning/Azure.Provisioning/api/Azure.Provisioning.net461.cs

```diff
@@ -116,7 +128,11 @@ public abstract partial class Resource<T> : Azure.Provisioning.Resource where T
     {
         protected Resource(Azure.Provisioning.IConstruct scope, Azure.Provisioning.Resource? parent, string resourceName, Azure.Core.ResourceType resourceType, string version, System.Func<string, T> createProperties, bool isExisting = false) : base (default(Azure.Provisioning.IConstruct), default(Azure.Provisioning.Resource), default(string), default(Azure.Core.ResourceType), default(string), default(System.Func<string, object>)) { }
         public T Properties { get { throw null; } }
+        public Azure.Provisioning.Output AddOutput(string outputName, Azure.Provisioning.BicepKind kind, System.Linq.Expressions.Expression<System.Func<T, object?>> propertySelector, bool isLiteral = false, bool isSecure = false) { throw null; }
```

Comment:
Could we automatically infer the type from the LINQ expression?

Reply:
There are some edge cases when mapping from bicep types to C# types, e.g. extensible enums.